src/process_utility.c

```diff
@@ -407,19 +414,59 @@ process_drop_trigger(DropStmt *stmt)
 	return handled;
 }
 
-static bool
+static void
+process_drop_index(DropStmt *stmt)
+{
+	ListCell   *lc;
+
+	foreach(lc, stmt->objects)
+	{
+		List	   *object = lfirst(lc);
+		RangeVar   *rv = makeRangeVarFromNameList(object);
+		Oid			idxrelid = RangeVarGetRelid(rv, NoLock, true);
+
+		if (OidIsValid(idxrelid))
+		{
+			Oid			tblrelid = IndexGetRelation(idxrelid, false);
+			Cache	   *hcache = hypertable_cache_pin();
+			Hypertable *ht = hypertable_cache_get_entry(hcache, tblrelid);
+
+			/*
+			 * Drop a hypertable index, i.e., all corresponding indexes on all
+			 * chunks
+			 */
+			if (NULL != ht)
+				chunk_index_delete_children_of(ht, idxrelid);
+			else
+			{
+				/* Drop an index on a chunk */
+				Chunk	   *chunk = chunk_get_by_relid(tblrelid, 0, false);
+
+				if (NULL != chunk)
+					chunk_index_delete(chunk, idxrelid);
+			}
+		}
+	}
+}
+
+static void
 process_drop(Node *parsetree)
 {
 	DropStmt   *stmt = (DropStmt *) parsetree;
 
 	switch (stmt->removeType)
 	{
 		case OBJECT_TABLE:
-			return process_drop_table(stmt);
+			process_drop_table(stmt);
```

Comment:
Why do you think this style is better? A return clearly indicates that nothing happens after....

Reply:
This function returns void now.